### PR TITLE
feat: amplify-sign-in component, authState refactor

### PR DESCRIPTION
**Pull request by jordanranz**
_Description of changes:_
- Sign-In Component using Amplify Auth category
- Refactored auth state management in Authenticator

[Delivers: #169195022]

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-ui-components/src/components/amplify-authenticator/__snapshots__/amplify-authenticator.spec.ts.snap [hunk 1]
```diff
@@ -2,9 +2,7 @@
 
 exports[`amplify-authenticator spec: Render logic -> should render a \`forgot password\` by passing \`forgotpassword\` into state prop 1`] = `
 <amplify-authenticator state="forgotpassword">
-  <div>
```
Comment (sammartinez) [inline packages/amplify-ui-components/src/components/amplify-authenticator/__snapshots__/amplify-authenticator.spec.ts.snap]: These look wrong.

File: packages/amplify-ui-components/src/components/amplify-authenticator/amplify-authenticator.tsx [hunk 2]
```diff
@@ -32,6 +32,7 @@ export class AmplifyAuthenticator {
         return <div>Loading...</div>;
       case AuthState.SignIn:
         return <amplify-sign-in handleAuthStateChange={this.onAuthStateChange} />;
+      // return <amplify-button>test</amplify-button>;
```
Comment (sammartinez) [inline packages/amplify-ui-components/src/components/amplify-authenticator/amplify-authenticator.tsx]: Is this needed?

File: packages/amplify-ui-components/src/components/amplify-authenticator/__snapshots__/amplify-authenticator.spec.ts.snap (hunk 1 above)
Comment (jordanranz) [reply packages/amplify-ui-components/src/components/amplify-authenticator/__snapshots__/amplify-authenticator.spec.ts.snap]: How so? `amplify-sign-in` has no props?

File: packages/amplify-ui-components/src/components/amplify-authenticator/amplify-authenticator.tsx (hunk 2 above)
Comment (jordanranz) [reply packages/amplify-ui-components/src/components/amplify-authenticator/amplify-authenticator.tsx]: Nope. good catch

File: packages/amplify-ui-components/src/components/amplify-authenticator/__snapshots__/amplify-authenticator.spec.ts.snap (hunk 1 above)
Comment (sammartinez) [reply packages/amplify-ui-components/src/components/amplify-authenticator/__snapshots__/amplify-authenticator.spec.ts.snap]: State that is passed in for this is forgotpassword, so the component that should be in place is the forgotpassword comp

File: packages/amplify-ui-components/src/components/amplify-sign-in/amplify-sign-in.tsx [hunk 3]
```diff
@@ -39,7 +33,9 @@ export class AmplifySignIn {
   @Prop() submitButtonText: string = SUBMIT_BUTTON_TEXT;
   /** (Optional) Overrides default styling */
   @Prop() overrideStyle: boolean = false;
-  /** 
+
```
Comment (sammartinez) [inline packages/amplify-ui-components/src/components/amplify-sign-in/amplify-sign-in.tsx]: Can we add a doc for this prop

**Pull request merged**